src/CodeStyle/Core/CodeFixes/FormattingCodeFixProvider.cs

```diff
@@ -13,11 +13,17 @@
 
 namespace Microsoft.CodeAnalysis.CodeStyle
 {
+    extern alias CodeStyle;
+    using Formatter = CodeStyle::Microsoft.CodeAnalysis.Formatting.Formatter;
+    using ISyntaxFormattingService = ISyntaxFormattingService;
```

Comment:
i have to admit, i find this totally confusing.  Things that were very weird to me.

1. You're both in a namespace called 'CodeStyle', but you have an extern-alias called 'CodeStyle'.
2. `using ISyntaxFormattingService = ISyntaxFormattingService`.  This is aliasing a type name available in that scope back to that name...

Reply:
i could use some clarification on the direction of how things are referenced.  I assume CodeStyle is not referencing Workspaces.  But i could imagine that Workspaces could reference CodeStyle and thus share a lot more code (instead of linking in copies).

Reply:
> using ISyntaxFormattingService = ISyntaxFormattingService

`cref` has different accessibility rules. The name is ambiguous without a using alias directive, but the directive itself benefits from normal code accessibility rules.